R/utils_render_latex.R

```diff
@@ -161,42 +161,55 @@ create_columns_component_l <- function(data) {
 #' @noRd
 create_body_component_l <- function(data) {
 
-  boxh <- dt_boxhead_get(data = data)
   body <- dt_body_get(data = data)
-  summaries_present <- dt_summary_exists(data = data)
+  boxh <- dt_boxhead_get(data = data)
+
   list_of_summaries <- dt_summary_df_get(data = data)
+  summaries_present <- dt_summary_exists(data = data)
   groups_rows_df <- dt_groups_rows_get(data = data)
   stub_components <- dt_stub_components(data = data)
 
+  # Obtain the `styles_tbl` (NOTE: this is currently unused)
   styles_tbl <- dt_styles_get(data = data)
 
   n_data_cols <- dt_boxhead_get_vars_default(data = data) %>% length()
   n_rows <- nrow(body)
 
-  # Get the column alignments for the data columns (this
-  # doesn't include the stub alignment)
-  col_alignment <- boxh[boxh$type == "default", ][["column_align"]]
+  # Determine whether the stub is available through analysis
+  # of the `stub_components`
+  stub_available <- dt_stub_components_has_rowname(stub_components = stub_components)
 
   # Get the column headings for the visible (e.g., `default`) columns
   default_vars <- dt_boxhead_get_vars_default(data = data)
 
-  if ("rowname" %in% names(body)) {
-    default_vars <- c("rowname", default_vars)
-  }
-
-  # Determine whether the stub is available through analysis
-  # of the `stub_components`
-  stub_available <- dt_stub_components_has_rowname(stub_components)
-
   if (stub_available) {
+
     n_cols <- n_data_cols + 1
+
+    stub_var <- dt_boxhead_get_var_stub(data = data)
+    all_stub_vals <- as.matrix(body[, stub_var])
+
+    body_content <- as.vector(t(cbind(all_stub_vals, body[, default_vars])))
+
   } else {
+
     n_cols <- n_data_cols
+
+    body_content <- as.vector(t(body[, default_vars]))
   }
 
-  # Get the sequence of column numbers in the table body (these
-  # are the visible columns in the table exclusive of the stub)
-  column_series <- seq(n_cols)
+  # Split `body_content` by slices of rows and create data rows
+  row_splits_body <-
+    split_body_content(
+      body_content = body_content,
+      n_cols = n_cols
+    )
+
+  body_rows <-
+    create_body_rows(
+      n_rows = n_rows,
+      row_splits_body = row_splits_body
+    )
```

Comment:
Could we derive `n_rows` from `length(row_splits_body)` instead of having them explicitly passed in (and potentially conflicting)?

Reply:
Now done.